### PR TITLE
[TP-274] physical stock module based on stockist stores

### PR DESCRIPTION
**Pull request by peter-rusin**

### CODE REVIEW

File: app/code/Convert/PhysicalStock/Block/PhysicalStock.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        \Magento\Framework\Registry $registry,
+        Template\Context $context,
+        array $data = []
+    ) {
+        parent::__construct($context, $data);
+        $this->config = $config;
+        $this->registry = $registry;
+    }
+
+    public function getConfig() : Config
```
Comment (bartekAtconvert) [inline app/code/Convert/PhysicalStock/Block/PhysicalStock.php:31]: please remember about proper PHPDocs. Have you got yet our coding guidelines? https://convert.atlassian.net/wiki/spaces/M2/pages/628097060/Coding+guidelines

File: app/code/Convert/PhysicalStock/etc/di.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <preference for="Convert\PhysicalStock\Api\Data\PhysicalStockInterface" type="Convert\PhysicalStock\Model\PhysicalStock"/>
+    <preference for="Convert\PhysicalStock\Api\PhysicalStockRepositoryInterface" type="Convert\PhysicalStock\Model\PhysicalStockRepository"/>
+    <type name="Convert\PhysicalStock\ViewModel\CurrentProduct">
```
Comment (bartekAtconvert) [inline app/code/Convert/PhysicalStock/etc/di.xml:6]: where is this CurrentProduct view model used?

File: app/code/Convert/PhysicalStock/view/frontend/layout/catalog_product_view.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (bartekAtconvert) [inline app/code/Convert/PhysicalStock/view/frontend/layout/catalog_product_view.xml:1]: - why Convert\PhysicalStock\Block\PhysicalStock is rendered 2 times?
- remember the requirements: " under “Add to Cart” button. Show only if the module enabled in Admin. "

File: app/code/Convert/PhysicalStock/view/frontend/templates/product/view/physical-stock.phtml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<?php /** @var $block \Convert\PhysicalStock\Block\PhysicalStock */ ?>
+<?php if ($block->getConfig()->isEnabled() && $block->getProduct()) : ?>
```
Comment (bartekAtconvert) [inline app/code/Convert/PhysicalStock/view/frontend/templates/product/view/physical-stock.phtml:2]: it's better to put this condition in layout

**Review by bartekAtconvert — CHANGES_REQUESTED** [review]

File: app/code/Convert/PhysicalStock/view/frontend/layout/catalog_product_view.xml (hunk 3 above)
Comment (peter-rusin) [reply app/code/Convert/PhysicalStock/view/frontend/layout/catalog_product_view.xml:1]: add to cart button is added twice - once for simple products and once for configurable, 

photo

I moved the new block to another `referenceContainer` which seems to be cleaner solution

File: app/code/Convert/PhysicalStock/etc/di.xml (hunk 2 above)
Comment (peter-rusin) [reply app/code/Convert/PhysicalStock/etc/di.xml:6]: leftovers, removed it

File: app/code/Convert/PhysicalStock/Block/PhysicalStock.php (hunk 1 above)
Comment (peter-rusin) [reply app/code/Convert/PhysicalStock/Block/PhysicalStock.php]: I'm not sure why this is wrong, coding guidelines state:
`9. A typed method signature SHOULD be preferred over PHPDoc annotations whenever possible.`

Comment (bartekAtconvert) [reply app/code/Convert/PhysicalStock/Block/PhysicalStock.php]: Good point

**Review by bartekAtconvert — APPROVED** [review]
much better now :-)

**Pull request merged**